android/libraries/rib-base/src/main/java/com/badoo/ribs/core/builder/BuildContext.kt

```diff
@@ -10,20 +11,23 @@ data class BuildContext internal constructor(
     val ancestryInfo: AncestryInfo,
     val attachMode: AttachMode = AttachMode.PARENT,
     val savedInstanceState: Bundle?,
-    val customisations: RibCustomisationDirectory
+    val customisations: RibCustomisationDirectory,
+    val plugins: List<Plugin> = emptyList()
```

Comment:
why you need it here? be careful with such kind of list in data class if you required somewhere equals state for it. because Plugins can be noncomparable

Reply:
If you check a bit further down, there's `plugins = rootPlugins`, only to be used from root creation to supply e.g. debugger plugin integration point without having to modify the internals of the rib itself. 

Otherwise constructor is internal.

Good point about equals, but thankfully BuildContext is not used as key and is not compared to another

Ended up in this ticket by accident from draft 1.0 branch though :)